src/components/UsingFeatureWrapper.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <FeatureWrapper
+    featureKey="YOUR-FEATURE-FLAG-KEY"
+    :userObject="userObject"
+    @flagValueChanged="handleFlagValueChange"
+  >
+    <TheNewFeature />
+    <template #else>
+      <!-- What you want to be displayed when the feature flag is turned off. You can add anything in this block like html elements or other vue components -->
+      <div class="feature-not-available-wrapper">
+        <p>Sorry this feature is not available. Your feature flag is off.</p>
+      </div>
+    </template>
+    <template #loading>
+      <!-- What you want to be displayed while the feature flag is loading. You can add anything in this block like html elements or other vue components -->
+      <div class="loading-wrapper">
+        <p>Loading...</p>
+      </div>
+    </template>
+  </FeatureWrapper>
+</template>
+
+<script setup lang="ts">
+import { inject, ref, onBeforeMount } from 'vue'
+
+// USING USER OBJECT: Import the User type
+import { User } from 'configcat-vue'
+
+// USING HOOKS: Import the ConfigCat client interface
+import type { IConfigCatClient } from 'configcat-vue'
+
+// USING HOOKS: Inject the underlying ConfigCat client
+const configCatClient = inject<IConfigCatClient>('configCatClient')
```

Comment:
nit: It'd be nice to check for `null`/`undefined` here as well like it's done in the UsingConfigCatClient example.

```suggestion
const configCatClient = inject<IConfigCatClient>('configCatClient')
  ?? (() => { throw new Error("ConfigCatPlugin was not installed."); })()
```